src/Components/WebAssembly/WebAssembly.Authentication/src/Models/InteractiveRequestOptions.cs

```diff
@@ -107,26 +125,27 @@ public override InteractiveRequestOptions Read(ref Utf8JsonReader reader, Type t
             };
         }
 
+        [UnconditionalSuppressMessage(
+            "Trimming",
+            "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code",
+            Justification = "This converter writes 'AdditionalRequestParameters' that might contain user defined types but that have already been annotated by 'TryAddAdditionalParameter'.")]
         public override void Write(Utf8JsonWriter writer, InteractiveRequestOptions value, JsonSerializerOptions options)
         {
             JsonSerializer.Serialize(
                 writer,
-                new OptionsRecord(value.ReturnUrl, value.Scopes, value.Interaction, value.AdditionalRequestParameters),
-                InteractiveRequestOptionsSerializerContext.Default.OptionsRecord);
+                new InteractiveOptions { ReturnUrl = value.ReturnUrl, Scopes = value.Scopes, Interaction = value.Interaction, AdditionalRequestParameters = value.AdditionalRequestParameters },
+                options);
         }
 
-        internal record struct OptionsRecord(
-            [property: JsonInclude] string ReturnUrl,
-            [property: JsonInclude] IEnumerable<string> Scopes,
-            [property: JsonInclude] InteractionType Interaction,
-            [property: JsonInclude] Dictionary<string, object> AdditionalRequestParameters);
-    }
-}
+        public struct InteractiveOptions
+        {
+            public string ReturnUrl { get; set; }
 
-[JsonSourceGenerationOptions(PropertyNamingPolicy = JsonKnownNamingPolicy.CamelCase, WriteIndented = false)]
-[JsonSerializable(typeof(InteractiveRequestOptions))]
-[JsonSerializable(typeof(InteractiveRequestOptions.Converter.OptionsRecord))]
-[JsonSerializable(typeof(JsonElement))]
-internal partial class InteractiveRequestOptionsSerializerContext : JsonSerializerContext
-{
+            public IEnumerable<string> Scopes { get; set; }
```

Comment:
Microsoft framework guidelines say you should use `ICollection` instead of `IEnumerable`.